webapp/views.py

```diff
@@ -60,8 +67,8 @@ def search(request):
 
     query = request.GET.get("q")
     site = request.GET.get("siteSearch") or request.GET.get("domain")
-    num = int(request.GET.get("num", "10"))
-    start = int(request.GET.get("start", "1"))
+    num = str_to_int(request.GET.get("num", 10), 10)
```

Comment:
So the question here is, what is the right thing to do when someone writes an erroneous URL like `/search?num=fish`.

Technically, I think we should probably be returning a [`400 Bad Request` status](https://developer.mozilla.org/en-US/docs/Web/HTTP/Status/400).

Reply:
Well here to "keep the experience" I thought that returning the default page without breaking the app.

Reply:
But who are these people going to `/search?num=asdf`? I think this is one of two cases:

- Someone's hacking on the URL in their browser
- HTML on some site is badly formed leading to the URL being badly formed

In both cases, I think it's more helpful to convey clearly that we didn't understand the request than to simply show them the first page of results. If they're hacking on the URL it's useful for them to know they got it wrong. If their website's HTML is malformed then they're probably not linking to the page they thought they were linking to, and it would be useful if that was made clear to them when they test clicking around their website.

This is an extension of the broader point I keep trying to make that I really don't think it's that helpful to silently handle errors, in basically any context.

Reply:
The current scenario, where https://ubuntu.com/search?q=kernel&num=10&start=asdf shows a 500, also doesn't seem to me to be the worst thing in the world. If doing the work to make it return the slightly more accurate 400 status seems like more trouble than it's worth, I would also be fine with leaving it as it is and simply "Ignore"ing the error in Sentry.